### PR TITLE
docs(defaults): list of default header values

### LINKED ISSUE #226: Document (or remove) user-agent header
It is not expected that the HTTP library is sending a user-agent header by default.

We only observed this when analysing the logs.

For others reference, `note-fetch` adds:

```
user-agent:       node-fetch/1.0 (+https://github.com/bitinn/node-fetch)
```

header to all requests, unless overwritten.

This can be a major privacy, security concern.

### PR DESCRIPTION
**Pull request by ahmadnassri**
Fixes #226

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -187,6 +187,18 @@ Note that only `method`, `headers`, `redirect` and `body` are allowed in `window
 }
 ```
 
+#### Default Headers
+
+If no values are set, the following request headers will be sent automatically:
+
+Header            | Value                                                   
+----------------- | --------------------------------------------------------
+`Accept-Encoding` | `gzip,deflate`
```
Comment (TimothyGu) [inline README.md]: This is done only when `compress` is true: https://github.com/bitinn/node-fetch/blob/560d5d6a023f069c37497f633fe65955182fa3d5/src/request.js#L145-L147

File: README.md [hunk 2]
```diff
@@ -187,6 +187,18 @@ Note that only `method`, `headers`, `redirect` and `body` are allowed in `window
...
+#### Default Headers
+
+If no values are set, the following request headers will be sent automatically:
+
+Header            | Value                                                   
+----------------- | --------------------------------------------------------
+`Accept-Encoding` | `gzip,deflate`                                          
+`Accept`          | `*/*`                                                   
+`Connection`      | `close`                                                 
+`Content-Length`  | (automatically calculated)
```
Comment (TimothyGu) [inline README.md]: (automatically calculated, if possible)

File: README.md [hunk 3]
```diff
@@ -187,6 +187,18 @@ Note that only `method`, `headers`, `redirect` and `body` are allowed in `window
...
 
+#### Default Headers
+
+If no values are set, the following request headers will be sent automatically:
+
+Header            | Value                                                   
+----------------- | --------------------------------------------------------
+`Accept-Encoding` | `gzip,deflate`                                          
+`Accept`          | `*/*`                                                   
+`Connection`      | `close`
```
Comment (TimothyGu) [inline README.md]: Specify that this is set automatically only if the user is not using an agent (https://github.com/bitinn/node-fetch/blob/560d5d6a023f069c37497f633fe65955182fa3d5/src/request.js#L148)

**Comment by ahmadnassri** [comment]
will make appropriate changes as suggested :+1:

File: README.md (hunk 3 above)
Comment (ahmadnassri) [inline README.md]: I feel that this is clearly implied (by declaring "default" in the section title)

Comment (TimothyGu) [inline README.md]: @ahmadnassri, no, the default only implies that it is set automatically in all cases when the header is not explicitly set. This header is special in that it is set automatically, when the header isn't explicitly set **and** `agent` isn't set.

Comment (ahmadnassri) [inline README.md]: I'm mistaken, somehow I conflated this with `User-Agent`! my bad.

**Pull request merged**